src/runtime/minrpc/minrpc_logger.cc

```diff
@@ -0,0 +1,307 @@
+  logger_.LogString("-> ERROR");
+  logger_.OutputLog();
+}
+
+void MinRPCReturnsWithLog::processValues(const TVMValue** values, const int** tcodes,
+                                         int* num_args) {
+  if (*tcodes != nullptr) {
+    logger_.LogString("-> [");
+    for (int i = 0; i < *num_args; ++i) {
+      logger_.LogTVMValue((*tcodes)[i], (*values)[i]);
+
+      if ((*tcodes)[i] == kTVMOpaqueHandle) {
+        this->registerHandleName((*values)[i].v_handle);
+      }
+    }
+    logger_.LogString("]");
+  }
+}
+
+void MinRPCReturnsWithLog::resetCurrHandleName(RPCCode code) {
+  this->code_ = code;
+  logger_.LogString(RPCCodeToString(code));
+  logger_.LogString(", ");
+  this->CurrHandleName_.clear();
+}
+
+void MinRPCReturnsWithLog::updateCurrHandleName(const char* name) {
+  if (this->CurrHandleName_.length() != 0) {
+    this->CurrHandleName_.append("::");
+  }
+  this->CurrHandleName_.append(name);
+}
+
+void MinRPCReturnsWithLog::getHandleName(void* handle) {
+  if (array_tracker_.find(handle) != array_tracker_.end()) {
+    CurrHandleName_.append(array_tracker_[handle]);
+    logger_.LogHandleName(CurrHandleName_);
+  }
+}
+
+void MinRPCReturnsWithLog::releaseHandleName(void* handle) {
+  if (array_tracker_.find(handle) != array_tracker_.end()) {
+    logger_.LogHandleName(array_tracker_[handle]);
+    array_tracker_.erase(handle);
+  }
+}
+
+void MinRPCReturnsWithLog::registerHandleName(void* handle) {
+  const std::string newString = CurrHandleName_;
+  array_tracker_[handle] = newString;
+}
+
+void MinRPCExecuteWithLog::ExecInitServer(int* _num_args) {
+  setRPCCode(RPCCode::kInitServer);
+  logger_->LogString("Init Server");
+  next_->ExecInitServer(_num_args);
+}
+
+void MinRPCExecuteWithLog::ExecNormalCallFunc(uint64_t* call_handle, TVMValue** values,
+                                              int** tcodes, int* num_args) {
+  setRPCCode(RPCCode::kCallFunc);
+  logger_->LogVal<void*>("call_handle: ", reinterpret_cast<void*>(*call_handle));
+  ret_handler_->getHandleName(reinterpret_cast<void*>(*call_handle));
+  if (*num_args > 0) {
+    logger_->LogString(", ");
+  }
+  this->processValues(values, tcodes, num_args);
+  next_->ExecNormalCallFunc(call_handle, values, tcodes, num_args);
+}
+
+void MinRPCExecuteWithLog::ExecCopyFromRemote(DLTensor** arr, uint64_t* num_bytes,
+                                              uint8_t** temp_data) {
+  setRPCCode(RPCCode::kCopyFromRemote);
+  logger_->LogVal<void*>("data_handle: ", reinterpret_cast<void*>((*arr)->data));
+  logger_->LogDLDevice(", DLDevice(type,id):", &(*arr)->device);
+  logger_->LogVal<int64_t>(", ndim: ", (*arr)->ndim);
+  logger_->LogDLData(", DLDataType(code,bits,lane): ", &(*arr)->dtype);
+  logger_->LogVal<uint64_t>(", num_bytes:", *num_bytes);
+  next_->ExecCopyFromRemote(arr, num_bytes, temp_data);
+}
+
+int MinRPCExecuteWithLog::ExecCopyToRemote(DLTensor** arr, uint64_t* _num_bytes,
+                                           uint8_t** _data_ptr) {
```

Comment:
Please don't use names that start with `_`.

Reply:
Thanks for pointing out to this, fixed